vm/systemSmartContracts/delegation.go

```diff
@@ -39,11 +39,15 @@ type delegation struct {
 	minServiceFee          uint64
 	maxServiceFee          uint64
 	minDelegationAmount    *big.Int
+	nodePrice              *big.Int
```

Comment:
Strongly suggest to add unit tests now for the new functionality from this file

Reply:
a lot of unit tests will come.